CMakeLists.txt

```diff
@@ -174,6 +174,24 @@ include(FetchContent)
 file(MAKE_DIRECTORY ${FETCHCONTENT_BASE_DIR}) # Ensure the directory exists
 message(STATUS "FetchContent base directory: ${FETCHCONTENT_BASE_DIR}")
 
+#
+# Set rocm version dev int.
+#
+if(VLLM_GPU_LANG STREQUAL "HIP")
+  #
+  # Setting up debug flags for pleasant debug experience.
+  #
+  set(CMAKE_${VLLM_GPU_LANG}_FLAGS_DEBUG "${CMAKE_${VLLM_GPU_LANG}_FLAGS_DEBUG} -O0 -ggdb3")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -ggdb3")
+
+
+  #
+  # Suppress the noisy warning
```

Comment:
Nit: could you add a more detailed comment along the lines of "Both CUDA and HIP device calls return an error code but they are marked [[noreturn]] only in HIP. This warning can be reenabled once we handle error codes from all device calls".